### PR TITLE
Bug 1264074 - Use Pulse for creation of Github resultsets

### PR DESCRIPTION
**Pull request by camd**
This adds the ability to ingest Github pushes and pull requests from the exchanges populated by the ``taskcluster-github`` project.

This follows a very similar pattern to the Pulse job ingestion:

* Adds a management command called ``read_pulse_resultsets`` which populates a queue called ``store_pulse_resultsets``.  Then the celery task will process those messages
* For each message, we need more information than is in the Pulse message (namely the revisions).  So we must call the Github api to get that information.

Still needed:

* taskcluster-github needs to include the base sha for pushes so I can get only the pertinent commits from github
* I will probably need to add a branch name to the ``Repository`` table so that I can distinguish which ``gaia`` or ``bugzilla`` repos it belongs to because they each share github repo urls, of course.

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1692)

### CODE REVIEW

**Comment by camd** [comment]
That's how we've been storing it in the revisions table thus far.  But I forgot the ``<`` and ``>``  surrounding the email.

**Comment by camd** [comment]
I don't want to make a schema change right now, tbh.  Will is in the process of moving from ``datasource`` to the ORM, and schema changes may make that cumbersome.

But perhaps after the change, we should consider splitting how we store that, yeah.

File: Procfile [hunk 1]
```diff
@@ -5,7 +5,9 @@ worker_buildapi_pending: newrelic-admin run-program celery -A treeherder worker
 worker_buildapi_running: newrelic-admin run-program celery -A treeherder worker -Q buildapi_running --maxtasksperchild=20 --concurrency=5
 worker_buildapi_4hr: newrelic-admin run-program celery -A treeherder worker -Q buildapi_4hr --maxtasksperchild=20 --concurrency=1
 worker_store_pulse_jobs: newrelic-admin run-program celery -A treeherder worker -Q store_pulse_jobs --maxtasksperchild=20 --concurrency=1
+worker_store_pulse_resultsets: newrelic-admin run-program celery -A treeherder worker -Q store_pulse_resultsets --maxtasksperchild=20 --concurrency=1
```
Comment (edmorley) [inline Procfile]: I know the `worker_store_pulse_jobs` entry also has a concurrency of 1, but I wonder if one or both can actually handle it being higher? So long as the RAM usage doesn't exceed the 512MG of the P1 dyno (and of course that CPU usage isn't so high that concurrency isn't gaining anything), then we could bump this up.

The current memory usage seems quite low:
https://dashboard.heroku.com/apps/treeherder-stage/metrics/worker_store_pulse_jobs

Comment (camd) [inline Procfile]: Sure, let's bump them to 3 for now and adjust as needed.  The traffic on the resultsets will be dramatically lower than for jobs, of course.  So I don't know if it matters much either way.  We don't even fall behind for jobs, afaict.

**Comment by camd** [comment]
I had to fix a couple things since the review:

* The order of the commits was coming in backwards.  I just had to swap them. Also fixed the tests to match.
* Added a couple logging lines to resultset ingestion in ``jobs.py`` to help keep track of how many were updated vs. inserted
* Upped the concurrency in the ``procfile``

**Comment by edmorley** [comment]
I've just realised the store_pulse_resultsets task doesn't retry at all - so if there was a failure eg connecting to github, the pushes will be lost. Could you make it a @retryable_task ?

**Pull request merged**